src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -303,4 +322,65 @@ public function addResource(ResourceInterface $resource)
             $this->resources[$key] = $resource;
         }
     }
+
+    /**
+     * Sets an alias for an existing route.
+     *
+     * @param string       $alias The alias to create
+     * @param string|Alias $route The route to alias
+     *
+     * @throws InvalidArgumentException if the id is not a string or an Alias
+     * @throws InvalidArgumentException if the alias is for itself
+     */
+    public function setAlias(string $alias, $route): Alias
+    {
+        if (\is_string($route)) {
+            $route = new Alias($route);
+        } elseif (!$route instanceof Alias) {
+            throw new InvalidArgumentException('$route must be a string, or an Alias object.');
+        }
+
+        if ($alias === (string) $route) {
+            throw new InvalidArgumentException(sprintf('An alias can not reference itself, got a circular reference on "%s".', $alias));
+        }
+
+        unset($this->routes[$alias], $this->priorities[$alias]);
+
+        return $this->aliases[$alias] = $route;
+    }
+
+    /**
+     * @return array<string, Alias>
+     */
+    public function getAliases(): array
+    {
+        return $this->aliases;
+    }
+
+    /**
+     * @return string[] An array of resolved route names
+     */
+    public function resolveAliases(): array
+    {
+        $resolvedAliases = [];
+        foreach ($this->aliases as $alias => $target) {
+            $resolved = $this->resolveAlias((string) $target);
+            if (null === $resolved) {
+                throw new RouteNotFoundException(sprintf('Target route "%s" for alias "%s" is not defined.', $target, $alias));
+            }
+
+            $resolvedAliases[$alias] = $resolved;
+        }
+
+        return $resolvedAliases;
+    }
+
+    private function resolveAlias(string $alias): ?string
```

Comment:
```suggestion

    /**
     * @return string|null 
     */
    private function resolveAlias(string $alias): ?string
```

Reply:
this is totally unnecessary, as the suggested phpdoc does not provide any additional info compared to the signature.